can't find libdbusmenu-glib-0.4 when build snap with gome-sdk

I’m trying to build an flutter app with gnome extension.
The `snapcraft.yaml` can be found in [here](https://github.com/gapplef/Fclash/blob/patch-1/snap/snapcraft.yaml).
when build the snap with github action I enounter the  following error:
```
:: [  +54 ms] CMake Error in flutter/ephemeral/.plugin_symlinks/tray_manager/linux/CMakeLists.txt:
:: [   +1 ms]   Imported target "PkgConfig::APPINDICATOR" includes non-existent path
:: [        ]     "/build/gnome-42-2204-sdk/stage/usr/include/libdbusmenu-glib-0.4"
:: [        ]   in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
:: [        ]   * The path was deleted, renamed, or moved to another location.
:: [        ]   * An install or uninstall procedure did not complete successfully.
:: [        ]   * The installation package was faulty and references files it does not
:: [        ]   provide.
:: [  +22 ms] -- Generating done
:: [        ] CMake Generate step failed.  Build files cannot be regenerated correctly.
:: [  +27 ms] Building Linux application... (completed in 3.2s)
:: [   +1 ms] "flutter linux" took 3,831ms.
:: [   +2 ms] Unable to generate build files
```
seems `libdbusmenu-glib-dev` is missing  in `gnome-sdk`.
I haved tried adding libdbusmenu-glib-dev to build-packages, and have no different.
I also tred to get help on the [snapcraft forum](https://forum.snapcraft.io/t/help-needed-build-snap-with-gnome-extension/34720/2) but get no reply.
Is this a bug of gnome-sdk? or how could I get this fixed? Any advice is appreciated!

> This is a gnome-42-2204-sdk bug, but not missing the dev package. I'll get this fixed soon.

Thanks a lot!, I'll try again once the new gnome-sdk relased.

> You'll also need to drop the build-packages for appindicator and harfbuzz, they are included in the sdk snap and will cause problems.

Without libayatana-appindicator I would get the following error.
```
:: [  +55 ms] CMake Error in flutter/ephemeral/.plugin_symlinks/tray_manager/linux/CMakeLists.txt:
:: [   +1 ms]   Imported target "PkgConfig::APPINDICATOR" includes non-existent path
:: [        ]     "/usr/include/libayatana-appindicator3-0.1"
:: [        ]   in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
:: [        ]   * The path was deleted, renamed, or moved to another location.
:: [        ]   * An install or uninstall procedure did not complete successfully.
:: [        ]   * The installation package was faulty and references files it does not
:: [        ]   provide.
:: [  +20 ms] -- Generating done
:: [        ] CMake Generate step failed.  Build files cannot be regenerated correctly.
:: [  +27 ms] Building Linux application... (completed in 3.4s)
:: [   +1 ms] "flutter linux" took 3,992ms.
:: [   +2 ms] Unable to generate build files
```
Adding build-packages for appindicator fix the problem. What might be the cause of the problem?

Really appreciated your help :smiling_face_with_three_hearts: :thumbsup:
I not sure if this is relevant. I encounter undefined reference error when build another apps with flutter snap. The apps can be build with `snapcraft.yaml` like the following(full file can be found [here](https://github.com/gapplef/tipitaka-pali-reader/blob/master/snap/snapcraft.yaml)):
```yaml
parts:
  tipikata-pali-reader:
    source: .
    plugin: nil
    override-build: |
      git clone --depth 1 -b stable https://github.com/flutter/flutter.git /root/parts/tipikata-pali-reader/build/flutter-distro
      export PATH="/root/parts/tipikata-pali-reader/build/flutter-distro/bin:$PATH"
      flutter precache --linux
      flutter pub get
      flutter build linux --release -v
    build-packages:
      - unzip
      - git
      - clang
      - cmake
```
It still  cannot managed to pack snap file due to some problems, but the building stage is exiting with code 0.
While when I try to use flutter snap instead of clone the source:
```yaml
parts:
  tipikata-pali-reader:
    source: .
    plugin: nil
    override-build: |
      flutter pub get
      flutter build linux --release -v
    build-snaps: [ flutter ]
```
I get the following error:
```
:: [ +178 ms] [7/10] Linking CXX shared library plugins/url_launcher_linux/liburl_launcher_linux_plugin.so
:: [  +12 ms] [8/10] Linking CXX shared library plugins/devicelocale/libdevicelocale_plugin.so
:: [ +407 ms] [9/10] Linking CXX executable intermediates_do_not_run/tipitaka_pali_reader
:: [        ] FAILED: intermediates_do_not_run/tipitaka_pali_reader
:: [  +13 ms] : && /snap/flutter/current/usr/bin/clang++  -O3 -DNDEBUG  -B/snap/flutter/current/usr/lib/gcc/x86_64-linux-gnu/9 -B/snap/flutter/current/usr/lib/x86_64-linux-gnu -B/snap/flutter/current/lib/x86_64-linux-gnu -B/snap/flutter/current/usr/lib/ -L/snap/flutter/current/usr/lib/gcc/x86_64-linux-gnu/9 -L/snap/flutter/current/usr/lib/x86_64-linux-gnu -L/snap/flutter/current/lib/x86_64-linux-gnu -L/snap/flutter/current/usr/lib/ -lblkid -lgcrypt -llzma -llz4 -lgpg-error -luuid -lpthread -ldl -lepoxy -B/snap/flutter/current/usr/lib/gcc/x86_64-linux-gnu/9 -B/snap/flutter/current/usr/lib/x86_64-linux-gnu -B/snap/flutter/current/lib/x86_64-linux-gnu -B/snap/flutter/current/usr/lib/ -L/snap/flutter/current/usr/lib/gcc/x86_64-linux-gnu/9 -L/snap/flutter/current/usr/lib/x86_64-linux-gnu -L/snap/flutter/current/lib/x86_64-linux-gnu -L/snap/flutter/current/usr/lib/ -lblkid -lgcrypt -llzma -llz4 -lgpg-error -luuid -lpthread -ldl -lepoxy CMakeFiles/tipitaka_pali_reader.dir/main.cc.o CMakeFiles/tipitaka_pali_reader.dir/my_application.cc.o CMakeFiles/tipitaka_pali_reader.dir/flutter/generated_plugin_registrant.cc.o  -o intermediates_do_not_run/tipitaka_pali_reader -L/root/parts/tipikata-pali-reader/build/linux/flutter/ephemeral -Wl,-rpath,/root/parts/tipikata-pali-reader/build/build/linux/x64/release/plugins/devicelocale:/root/parts/tipikata-pali-reader/build/build/linux/x64/release/plugins/url_launcher_linux:/root/parts/tipikata-pali-reader/build/linux/flutter/ephemeral:  plugins/devicelocale/libdevicelocale_plugin.so  plugins/url_launcher_linux/liburl_launcher_linux_plugin.so  -lflutter_linux_gtk  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libgtk-3.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libgdk-3.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libpango-1.0.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libharfbuzz.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libatk-1.0.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libcairo-gobject.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libcairo.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libgio-2.0.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libgobject-2.0.so  /snap/flutter/current/usr/lib/x86_64-linux-gnu/libglib-2.0.so && :
:: [        ] /snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libsystemd.so.0: undefined reference to `mq_getattr@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libmount.so.1: undefined reference to `fstatat@GLIBC_2.33'
:: [   +1 ms] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0: undefined reference to `pango_font_family_get_face'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0: undefined reference to `dlopen@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0: undefined reference to `pango_matrix_get_slant_ratio'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libxkbcommon.so.0: undefined reference to `fstat64@GLIBC_2.33'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0: undefined reference to `pango_font_get_languages'
:: [        ] /snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libsystemd.so.0: undefined reference to `pthread_sigmask@GLIBC_2.32'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/lib/x86_64-linux-gnu/libdbus-1.so.3: undefined reference to `pthread_mutexattr_init@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libsystemd.so.0: undefined reference to `fstatat64@GLIBC_2.33'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/lib/x86_64-linux-gnu/libdbus-1.so.3: undefined reference to `pthread_condattr_setclock@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0: undefined reference to `dlerror@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libxkbcommon.so.0: undefined reference to `stat64@GLIBC_2.33'
:: [        ] /snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libsystemd.so.0: undefined reference to `close_range@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/lib/x86_64-linux-gnu/libdbus-1.so.3: undefined reference to `pthread_mutexattr_destroy@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libfontconfig.so.1: undefined reference to `lstat@GLIBC_2.33'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/lib/x86_64-linux-gnu/libdbus-1.so.3: undefined reference to `pthread_mutexattr_settype@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libsystemd.so.0: undefined reference to `pthread_join@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0: undefined reference to `dlclose@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libsystemd.so.0: undefined reference to `pthread_setname_np@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0: undefined reference to `hb_font_set_synthetic_slant'
:: [   +1 ms] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0: undefined reference to `pango_font_map_get_family'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libX11.so.6: undefined reference to `fstat@GLIBC_2.33'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libX11.so.6: undefined reference to `stat@GLIBC_2.33'
:: [        ] /snap/flutter/current/usr/bin/ld: /lib/x86_64-linux-gnu/libsystemd.so.0: undefined reference to `pthread_once@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0: undefined reference to `dlsym@GLIBC_2.34'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0: undefined reference to `g_memdup2'
:: [        ] /snap/flutter/current/usr/bin/ld: /snap/gnome-42-2204-sdk/current/lib/x86_64-linux-gnu/libselinux.so.1: undefined reference to `lstat64@GLIBC_2.33'
:: [        ] clang: error: linker command failed with exit code 1 (use -v to see invocation)
:: [   +1 ms] ninja: build stopped: subcommand failed.
:: [  +11 ms] Building Linux application... (completed in 67.2s)
:: [   +1 ms] "flutter linux" took 67,860ms.
:: [   +2 ms] Build process failed
:: [        ]
::            #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
::            #1      _runBuild (package:flutter_tools/src/linux/build_linux.dart:183:5)
```
I not sure is this problem related to gnome-sdk or flutter-snap. Any advice?

--- a/snapcraft.yaml
+++ b/snapcraft.yaml
@@ -1318,9 +1318,18 @@ parts:
       - usr/share/gir-1.0/Aya*
       - usr/share/vala/aya*
       - usr/include/ayata*
-      - usr/lib/*/pkgconfig
+      - usr/lib/*/pkgconfig/*.pc
       - usr/lib/*/*ido*
       - usr/lib/*/girepository-1.0/Ayatana*
+    override-build: |
+      set -eux
+      craftctl default
+      for PC in $CRAFT_PART_INSTALL/usr/lib/$CRAFT_ARCH_TRIPLET/pkgconfig/*.pc;
+      do
+        sed -i 's#exec_prefix=/usr#exec_prefix=${prefix}#' $PC
+        sed -i 's#libdir=/usr#libdir=${prefix}#' $PC
+        sed -i 's#includedir=/usr#includedir=${prefix}#' $PC
+      done
 
   libayatana-appindicator:
     after: [libayatana-ido, glib, gtk3]
@@ -1355,15 +1364,24 @@ parts:
       - usr/share/vala/vapi/Dbusmenu*
       - usr/include/*ayata*
       - usr/include/*dbusmenu*
-      - usr/lib/*/pkgconfig/*indica*
-      - usr/lib/*/pkgconfig/*dbusmenu*
+      - usr/lib/*/pkgconfig/*indica*.pc
+      - usr/lib/*/pkgconfig/*dbusmenu*.pc
       - usr/lib/*/*indica*
       - usr/lib/*/*dbusmenu*
       - usr/lib/*/girepository-1.0/AyatanaAppIndicator*
       - usr/lib/*/girepository-1.0/Dbusmenu*
+    override-build: |
+      set -eux
+      craftctl default
+      for PC in $CRAFT_PART_INSTALL/usr/lib/$CRAFT_ARCH_TRIPLET/pkgconfig/*.pc;
+      do
+        sed -i 's#exec_prefix=/usr#exec_prefix=${prefix}#' $PC
+        sed -i 's#libdir=/usr#libdir=${prefix}#' $PC
+        sed -i 's#includedir=/usr#includedir=${prefix}#' $PC
+      done
 
   debs:
-    after: [ libgnome-games-support, libadwaita, libgweather, poppler ]
+    after: [ libgnome-games-support, libadwaita, libgweather, poppler, libayatana-appindicator ]
     plugin: nil
     stage-packages:
       - appstream

